Sources/SmokeHTTPClient/HTTPClient+executeSyncRetriableWithoutOutput.swift

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+//  HTTPClient+executeSyncRetriableWithoutOutput.swift
+//  SmokeHTTPClient
+//
+
+import Foundation
+import NIO
+import NIOHTTP1
+import NIOOpenSSL
+import NIOTLS
+import LoggerAPI
+
+public extension HTTPClient {
+    /**
+     Helper type that manages the state of a retriable sync request.
+     */
+    private class ExecuteSyncWithoutOutputRetriable<InputType>
+            where InputType: HTTPRequestInputProtocol {
+        let endpointOverride: URL?
+        let endpointPath: String
+        let httpMethod: HTTPMethod
+        let input: InputType
+        let handlerDelegate: HTTPClientChannelInboundHandlerDelegate
+        let httpClient: HTTPClient
+        let retryConfiguration: HTTPClientRetryConfiguration
+        
+        var retriesRemaining: Int
+        
+        let milliToMicroSeconds = 1000
```

Comment:
This would be clearer as a function/closure.

Reply:
I'm not sure what you mean here. Could you elaborate?

Reply:
Oh yes. That would be clearer. I will make that change.